xilem/examples/virtual_cats.rs

```diff
@@ -71,50 +75,74 @@ impl VirtualCats {
+                    let Some(status) = state.statuses.iter_mut().find(|it| it.code == code) else {
                         unreachable!("We never remove items from `statuses`")
+                    };
+                    match message {
+                        Ok(image) => {
+                            status.image = ImageState::Available(image);
+                        }
+                        Err(err) => status.image = ImageState::Error(err),
                     }
                 },
             ))
         } else {
             None
         };
-        let img = if let Some(img) = img {
-            let attribution = sized_box(
-                sized_box(
-                    prose("Copyright ©️ https://http.cat")
-                        .line_break_mode(LineBreaking::Clip)
-                        .text_alignment(TextAlign::End),
+        let img = match &item.image {
+            ImageState::Available(img) => {
+                let attribution = sized_box(
+                    sized_box(
+                        prose("Copyright ©️ https://http.cat")
+                            .line_break_mode(LineBreaking::Clip)
+                            .text_alignment(TextAlign::End),
+                    )
+                    .padding(4.)
+                    .corner_radius(4.)
+                    .background_color(palette::css::BLACK.multiply_alpha(0.5)),
                 )
-                .padding(4.)
-                .corner_radius(4.)
-                .background_color(palette::css::BLACK.multiply_alpha(0.5)),
-            )
-            .padding(Padding {
-                left: 0.,
-                right: 42.,
-                top: 30.,
-                bottom: 0.,
-            });
-            Either::A(zstack((
-                image(img).fit(ObjectFit::FitWidth),
-                attribution.alignment(UnitPoint::TOP_RIGHT),
-            )))
-        } else {
-            Either::B(sized_box(spinner()).width(80.px()).height(80.px()))
+                .padding(Padding {
+                    left: 0.,
+                    right: 42.,
+                    top: 30.,
+                    bottom: 0.,
+                });
+                let imgview = zstack((
+                    image(img).fit(ObjectFit::FitWidth),
+                    attribution.alignment(UnitPoint::TOP_RIGHT),
+                ));
+                OneOf3::A(imgview)
+            }
+            ImageState::Pending => OneOf3::B(sized_box(spinner()).width(80.px()).height(80.px())),
+            ImageState::Error(err) => {
+                // the people deserve their cat.
+                // It is vital that the cat explains what went wrong.
+                let asciicat = label(
+                    "
+                  /\\_/\\  !
+                 ( o.o )
+                  > ^ <",
+                )
+                .color(BLACK);
+                let errorstring = prose(err.to_string()).text_color(Color::from_rgb8(255, 0, 0));
```

Comment:
This text color/background combination possibly isn't sufficiently high contrast:
https://webaim.org/resources/contrastchecker/?fcolor=FF0000&bcolor=FFFFFF (I'm not certain this is the best website, but it gets the point across).
It's at least close, so it might be fine? Maybe we should make this text slightly larger or bold?